mathics/builtin/patterns.py

```diff
@@ -676,35 +750,43 @@ def yield_match(vars_2, rest):
             ):
                 yield_func(vars_2, None)
 
-        self.pattern.match(yield_match, expression, vars, evaluation)
+        pattern_context = pattern_context.copy()
+        pattern_context["yield_func"] = yield_match
+        self.pattern.match(expression, pattern_context)
 
     def quick_pattern_test(self, candidate, test, evaluation: Evaluation):
+        """Pattern test for some other special cases"""
         if test == "System`NegativePowerQ":
             return (
                 candidate.has_form("Power", 2)
                 and isinstance(candidate.elements[1], (Integer, Rational, Real))
                 and candidate.elements[1].value < 0
             )
-        elif test == "System`NotNegativePowerQ":
+        if test == "System`NotNegativePowerQ":
             return not (
                 candidate.has_form("Power", 2)
                 and isinstance(candidate.elements[1], (Integer, Rational, Real))
                 and candidate.elements[1].value < 0
             )
-        else:
-            from mathics.core.builtin import Test
-
-            builtin = None
-            builtin = evaluation.definitions.get_definition(test)
-            if builtin:
-                builtin = builtin.builtin
-            if builtin is not None and isinstance(builtin, Test):
-                return builtin.test(candidate)
+
+        from mathics.core.builtin import Test
```

Comment:
This can be moved to the top of the file and will make our linter overlords happy. Will be in the next commit of more_patterns_more